src/tests/JIT/Generics/VirtualMethods/generic_virtual_methods.cs

```diff
@@ -0,0 +1,724 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
+
+using System;
+using System.Runtime.CompilerServices;
+using Xunit;
+
+public class GenericVirtualMethodTests
+{
+    [Fact]
```

Comment:
Make each of these categories be the `[Fact]`s so if something fails we have a bit more granularity.